app/renderer/js/pages/preference/general-section.ts

```diff
@@ -389,11 +390,78 @@ export default class GeneralSection extends BaseSection {
 		}
 	}
 
-	updateResetDataOption(): void {
-		const resetDataButton = document.querySelector('#resetdata-option .reset-data-button');
+	/* Reset App Data */
+	resetAppData(): void {
+		const resetAppButton = document.querySelector('#resetdata-option .reset-data-button');
+		resetAppButton.addEventListener('click', () => {
+			this.resetAppSettings();
+		});
+	}
+
+	async resetAppSettings(): Promise<void> {
+		const resetMessage = 'Selecting YES will reset all the settings to default without removing any connected organizations, certificates and accounts from Zulip.';
+		// We save the Application's settings/configurations in following files
+		// Preferences file is able to reset zoom levels but requires restart, not hard reload
+		const settingFiles = ['config/settings.json', 'config/window-state.json', 'Partitions/webviewsession/Preferences'];
+		const { response } = await dialog.showMessageBox({
+			type: 'warning',
+			buttons: ['YES', 'NO'],
+			defaultId: 0,
+			message: 'Are you sure?',
+			detail: resetMessage
+		});
+		if (response === 0) {
+			settingFiles.forEach(settingFileName => {
+				const getSettingFilesPath = path.join(app.getPath('appData'), appName, settingFileName);
+				fs.access(getSettingFilesPath, (error: any) => {
+					if (error) {
+						logger.error('Error while resetting app settings.');
+						logger.error(error);
+					} else {
+						fs.unlink(getSettingFilesPath, () => {
+							this.sendAction('clear-app-data');
+						});
+					}
+				});
+			});
+			setTimeout(() => ipcRenderer.send('forward-message', 'hard-reload'), 1000);
+		}
+	}
+	/* // Reset App Data */
+
+	/* Factory Reset */
+	factoryReset(): void {
+		const resetDataButton = document.querySelector('#factory-reset-option .factory-reset-button');
 		resetDataButton.addEventListener('click', () => {
-			this.clearAppDataDialog();
+			this.factoryResetSettings();
+		});
+	}
+
+	async factoryResetSettings(): Promise<void> {
+		const clearAppDataMessage = 'Selecting YES will remove all organisations, certificates, accounts and preferences from Zulip.\nWhen the application restarts, it will be as if you are starting Zulip for the first time.';
```

Comment:
Please spell "organizations"; we use American english spelling.

Reply:
Will push in next commit :)